From foo@baz Wed Aug 19 16:14:47 PDT 2009
Date: Wed, 19 Aug 2009 16:14:47 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: move osd.h

This moves osd.h out of the include/ subdirectory.

No code changes are made here.

Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/BlkVsc.c           |    1 
 drivers/staging/hv/Channel.c          |    2 
 drivers/staging/hv/Channel.h          |    1 
 drivers/staging/hv/ChannelInterface.c |    3 -
 drivers/staging/hv/ChannelMgmt.c      |    2 
 drivers/staging/hv/ChannelMgmt.h      |    1 
 drivers/staging/hv/Connection.c       |    1 
 drivers/staging/hv/Hv.c               |    1 
 drivers/staging/hv/Hv.h               |    2 
 drivers/staging/hv/NetVsc.c           |    1 
 drivers/staging/hv/RingBuffer.c       |    1 
 drivers/staging/hv/RingBuffer.h       |    2 
 drivers/staging/hv/RndisFilter.c      |    2 
 drivers/staging/hv/RndisFilter.h      |    1 
 drivers/staging/hv/StorVsc.c          |    2 
 drivers/staging/hv/Vmbus.c            |    1 
 drivers/staging/hv/blkvsc_drv.c       |    1 
 drivers/staging/hv/include/List.h     |    1 
 drivers/staging/hv/include/VmbusApi.h |    1 
 drivers/staging/hv/include/logging.h  |    1 
 drivers/staging/hv/include/osd.h      |   78 ----------------------------------
 drivers/staging/hv/netvsc_drv.c       |    1 
 drivers/staging/hv/osd.c              |    2 
 drivers/staging/hv/osd.h              |   78 ++++++++++++++++++++++++++++++++++
 drivers/staging/hv/storvsc_drv.c      |    2 
 drivers/staging/hv/vmbus_drv.c        |    2 
 26 files changed, 94 insertions(+), 97 deletions(-)

--- a/drivers/staging/hv/BlkVsc.c
+++ b/drivers/staging/hv/BlkVsc.c
@@ -22,6 +22,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
+#include "osd.h"
 #include "StorVsc.c"
 
 static const char* gBlkDriverName="blkvsc";
--- a/drivers/staging/hv/blkvsc_drv.c
+++ b/drivers/staging/hv/blkvsc_drv.c
@@ -33,6 +33,7 @@
 #include <scsi/scsi_eh.h>
 #include <scsi/scsi_dbg.h>
 
+#include "osd.h"
 #include "include/logging.h"
 #include "include/vmbus.h"
 
--- a/drivers/staging/hv/Channel.c
+++ b/drivers/staging/hv/Channel.c
@@ -23,7 +23,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
-#include "include/osd.h"
+#include "osd.h"
 #include "include/logging.h"
 
 #include "VmbusPrivate.h"
--- a/drivers/staging/hv/Channel.h
+++ b/drivers/staging/hv/Channel.h
@@ -25,7 +25,6 @@
 #ifndef _CHANNEL_H_
 #define _CHANNEL_H_
 
-#include "include/osd.h"
 #include "ChannelMgmt.h"
 
 /* The format must be the same as VMDATA_GPA_DIRECT */
--- a/drivers/staging/hv/ChannelInterface.c
+++ b/drivers/staging/hv/ChannelInterface.c
@@ -23,8 +23,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
-#include "include/osd.h"
-
+#include "osd.h"
 #include "VmbusPrivate.h"
 
 static int
--- a/drivers/staging/hv/ChannelMgmt.c
+++ b/drivers/staging/hv/ChannelMgmt.c
@@ -24,7 +24,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
-#include "include/osd.h"
+#include "osd.h"
 #include "include/logging.h"
 
 #include "VmbusPrivate.h"
--- a/drivers/staging/hv/ChannelMgmt.h
+++ b/drivers/staging/hv/ChannelMgmt.h
@@ -25,7 +25,6 @@
 #ifndef _CHANNEL_MGMT_H_
 #define _CHANNEL_MGMT_H_
 
-#include "include/osd.h"
 #include "include/List.h"
 #include "RingBuffer.h"
 
--- a/drivers/staging/hv/Connection.c
+++ b/drivers/staging/hv/Connection.c
@@ -25,6 +25,7 @@
 #include <linux/kernel.h>
 #include <linux/mm.h>
 #include <linux/vmalloc.h>
+#include "osd.h"
 #include "include/logging.h"
 #include "VmbusPrivate.h"
 
--- a/drivers/staging/hv/Hv.c
+++ b/drivers/staging/hv/Hv.c
@@ -24,6 +24,7 @@
 #include <linux/kernel.h>
 #include <linux/mm.h>
 #include <linux/vmalloc.h>
+#include "osd.h"
 #include "include/logging.h"
 #include "VmbusPrivate.h"
 
--- a/drivers/staging/hv/Hv.h
+++ b/drivers/staging/hv/Hv.h
@@ -25,8 +25,6 @@
 #ifndef __HV_H__
 #define __HV_H__
 
-#include "include/osd.h"
-
 #include "include/HvTypes.h"
 #include "include/HvStatus.h"
 #include "include/HvHalApi.h"
--- a/drivers/staging/hv/include/List.h
+++ b/drivers/staging/hv/include/List.h
@@ -25,7 +25,6 @@
 #ifndef _LIST_H_
 #define _LIST_H_
 
-#include "osd.h"
 /*
  *
  *  Doubly-linked list manipulation routines.  Implemented as macros
--- a/drivers/staging/hv/include/logging.h
+++ b/drivers/staging/hv/include/logging.h
@@ -28,7 +28,6 @@
 /* #include <linux/init.h> */
 /* #include <linux/module.h> */
 
-#include "osd.h"
 
 #define VMBUS				0x0001
 #define STORVSC				0x0002
--- a/drivers/staging/hv/include/osd.h
+++ /dev/null
@@ -1,78 +0,0 @@
-/*
- *
- * Copyright (c) 2009, Microsoft Corporation.
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms and conditions of the GNU General Public License,
- * version 2, as published by the Free Software Foundation.
- *
- * This program is distributed in the hope it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
- * Place - Suite 330, Boston, MA 02111-1307 USA.
- *
- * Authors:
- *   Haiyang Zhang <haiyangz@microsoft.com>
- *   Hank Janssen  <hjanssen@microsoft.com>
- *
- */
-
-
-#ifndef _OSD_H_
-#define _OSD_H_
-
-
-/* Defines */
-
-
-
-#define ALIGN_UP(value, align)			( ((value) & (align-1))? ( ((value) + (align-1)) & ~(align-1) ): (value) )
-#define ALIGN_DOWN(value, align)		( (value) & ~(align-1) )
-#define NUM_PAGES_SPANNED(addr, len)	( (ALIGN_UP(addr+len, PAGE_SIZE) - ALIGN_DOWN(addr, PAGE_SIZE)) >> PAGE_SHIFT )
-
-#define LOWORD(dw)		((unsigned short) (dw))
-#define HIWORD(dw)		((unsigned short) (((unsigned int) (dw) >> 16) & 0xFFFF))
-
-typedef struct _DLIST_ENTRY {
-   struct _DLIST_ENTRY *Flink;
-   struct _DLIST_ENTRY *Blink;
-} DLIST_ENTRY;
-
-
-/* Other types */
-
-/* typedef unsigned char		GUID[16]; */
-
-typedef struct {
-	unsigned char	Data[16];
-} GUID;
-
-struct osd_waitevent {
-	int	condition;
-	wait_queue_head_t event;
-};
-
-/* Osd routines */
-
-extern void *osd_VirtualAllocExec(unsigned int size);
-
-extern void *osd_PageAlloc(unsigned int count);
-extern void osd_PageFree(void* page, unsigned int count);
-
-extern struct osd_waitevent *osd_WaitEventCreate(void);
-extern void osd_WaitEventSet(struct osd_waitevent *waitEvent);
-extern int osd_WaitEventWait(struct osd_waitevent *waitEvent);
-
-/* If >0, waitEvent got signaled. If ==0, timeout. If < 0, error */
-extern int osd_WaitEventWaitEx(struct osd_waitevent *waitEvent, u32 TimeoutInMs);
-
-
-int osd_schedule_callback(struct workqueue_struct *wq,
-			  void (*func)(void *),
-			  void *data);
-
-#endif /* _OSD_H_ */
--- a/drivers/staging/hv/include/VmbusApi.h
+++ b/drivers/staging/hv/include/VmbusApi.h
@@ -25,7 +25,6 @@
 #ifndef _VMBUS_API_H_
 #define _VMBUS_API_H_
 
-#include "osd.h"
 
 
 /* Defines */
--- a/drivers/staging/hv/NetVsc.c
+++ b/drivers/staging/hv/NetVsc.c
@@ -24,6 +24,7 @@
 #include <linux/mm.h>
 #include <linux/delay.h>
 #include <asm/io.h>
+#include "osd.h"
 #include "include/logging.h"
 #include "NetVsc.h"
 #include "RndisFilter.h"
--- a/drivers/staging/hv/netvsc_drv.c
+++ b/drivers/staging/hv/netvsc_drv.c
@@ -36,6 +36,7 @@
 #include <net/sock.h>
 #include <net/pkt_sched.h>
 
+#include "osd.h"
 #include "include/logging.h"
 #include "include/vmbus.h"
 
--- a/drivers/staging/hv/osd.c
+++ b/drivers/staging/hv/osd.c
@@ -43,7 +43,7 @@
 #include <asm/kmap_types.h>
 #include <asm/atomic.h>
 
-#include "include/osd.h"
+#include "osd.h"
 
 
 /* Data types */
--- /dev/null
+++ b/drivers/staging/hv/osd.h
@@ -0,0 +1,78 @@
+/*
+ *
+ * Copyright (c) 2009, Microsoft Corporation.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place - Suite 330, Boston, MA 02111-1307 USA.
+ *
+ * Authors:
+ *   Haiyang Zhang <haiyangz@microsoft.com>
+ *   Hank Janssen  <hjanssen@microsoft.com>
+ *
+ */
+
+
+#ifndef _OSD_H_
+#define _OSD_H_
+
+
+/* Defines */
+
+
+
+#define ALIGN_UP(value, align)			( ((value) & (align-1))? ( ((value) + (align-1)) & ~(align-1) ): (value) )
+#define ALIGN_DOWN(value, align)		( (value) & ~(align-1) )
+#define NUM_PAGES_SPANNED(addr, len)	( (ALIGN_UP(addr+len, PAGE_SIZE) - ALIGN_DOWN(addr, PAGE_SIZE)) >> PAGE_SHIFT )
+
+#define LOWORD(dw)		((unsigned short) (dw))
+#define HIWORD(dw)		((unsigned short) (((unsigned int) (dw) >> 16) & 0xFFFF))
+
+typedef struct _DLIST_ENTRY {
+   struct _DLIST_ENTRY *Flink;
+   struct _DLIST_ENTRY *Blink;
+} DLIST_ENTRY;
+
+
+/* Other types */
+
+/* typedef unsigned char		GUID[16]; */
+
+typedef struct {
+	unsigned char	Data[16];
+} GUID;
+
+struct osd_waitevent {
+	int	condition;
+	wait_queue_head_t event;
+};
+
+/* Osd routines */
+
+extern void *osd_VirtualAllocExec(unsigned int size);
+
+extern void *osd_PageAlloc(unsigned int count);
+extern void osd_PageFree(void* page, unsigned int count);
+
+extern struct osd_waitevent *osd_WaitEventCreate(void);
+extern void osd_WaitEventSet(struct osd_waitevent *waitEvent);
+extern int osd_WaitEventWait(struct osd_waitevent *waitEvent);
+
+/* If >0, waitEvent got signaled. If ==0, timeout. If < 0, error */
+extern int osd_WaitEventWaitEx(struct osd_waitevent *waitEvent, u32 TimeoutInMs);
+
+
+int osd_schedule_callback(struct workqueue_struct *wq,
+			  void (*func)(void *),
+			  void *data);
+
+#endif /* _OSD_H_ */
--- a/drivers/staging/hv/RingBuffer.c
+++ b/drivers/staging/hv/RingBuffer.c
@@ -23,6 +23,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
+#include "osd.h"
 #include "include/logging.h"
 #include "RingBuffer.h"
 
--- a/drivers/staging/hv/RingBuffer.h
+++ b/drivers/staging/hv/RingBuffer.h
@@ -27,8 +27,6 @@
 
 #include <linux/scatterlist.h>
 
-#include "include/osd.h"
-
 typedef struct _RING_BUFFER {
 	/* Offset in bytes from the start of ring data below */
 	volatile u32 WriteIndex;
--- a/drivers/staging/hv/RndisFilter.c
+++ b/drivers/staging/hv/RndisFilter.c
@@ -26,8 +26,8 @@
 #include <asm/kmap_types.h>
 #include <asm/io.h>
 
+#include "osd.h"
 #include "include/logging.h"
-
 #include "include/NetVscApi.h"
 #include "RndisFilter.h"
 
--- a/drivers/staging/hv/RndisFilter.h
+++ b/drivers/staging/hv/RndisFilter.h
@@ -27,7 +27,6 @@
 
 #define __struct_bcount(x)
 
-#include "include/osd.h"
 #include "NetVsc.h"
 
 #include "include/rndis.h"
--- a/drivers/staging/hv/StorVsc.c
+++ b/drivers/staging/hv/StorVsc.c
@@ -25,8 +25,8 @@
 #include <linux/string.h>
 #include <linux/mm.h>
 #include <linux/delay.h>
+#include "osd.h"
 #include "include/logging.h"
-
 #include "include/StorVscApi.h"
 #include "include/VmbusPacketFormat.h"
 #include "include/vstorage.h"
--- a/drivers/staging/hv/storvsc_drv.c
+++ b/drivers/staging/hv/storvsc_drv.c
@@ -36,9 +36,9 @@
 
 #include <scsi/scsi_dbg.h>
 
+#include "osd.h"
 #include "include/logging.h"
 #include "include/vmbus.h"
-
 #include "include/StorVscApi.h"
 
 
--- a/drivers/staging/hv/Vmbus.c
+++ b/drivers/staging/hv/Vmbus.c
@@ -23,6 +23,7 @@
 
 #include <linux/kernel.h>
 #include <linux/mm.h>
+#include "osd.h"
 #include "include/logging.h"
 #include "VersionInfo.h"
 #include "VmbusPrivate.h"
--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -28,7 +28,7 @@
 #include <linux/irq.h>
 #include <linux/interrupt.h>
 #include <linux/sysctl.h>
-
+#include "osd.h"
 #include "include/logging.h"
 #include "include/vmbus.h"
 
